src/Parser/Monk/Windwalker/Modules/Items/TheEmperorsCapacitor.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellLink from 'common/SpellLink';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Module from 'Parser/Core/Module';
+
+import CHI_SPENDERS from 'Parser/Monk/Windwalker/Constants';
+
+class TheEmperorsCapacitor extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  totalStacks = 0;
+  currentStacks = 0;
+  stacksUsed = 0;
+  stacksWasted = 0;
+  cjlCasts = 0;
+  averageStacksUsed = 0;
+  damage = 0;
+  chiSpenders = CHI_SPENDERS;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasChest(ITEMS.THE_EMPERORS_CAPACITOR.id);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    // First stack of the buff
+    if (SPELLS.THE_EMPERORS_CAPACITOR_STACK.id === spellId) {
+      this.totalStacks += 1;
+      this.currentStacks += 1;
+    }
+  }
+
+  on_byPlayer_applybuffstack(event) {
+    const spellId = event.ability.guid;
+    // You can only ever apply 1 stack at a time
+    if (SPELLS.THE_EMPERORS_CAPACITOR_STACK.id === spellId) {
+      this.totalStacks += 1;     
+      this.currentStacks += 1;
+     }
+   }
+  
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (this.chiSpenders.includes(spellId) && this.currentStacks === 20) {
+      this.stacksWasted += 1;
+    }
+
+    if (SPELLS.CRACKLING_JADE_LIGHTNING.id === spellId) {
+      this.stacksUsed += this.currentStacks;
+      this.currentStacks = 0;
+      this.cjlCasts += 1;
+      this.averageStacksUsed = this.stacksUsed / this.cjlCasts;
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId === SPELLS.CRACKLING_JADE_LIGHTNING.id) {
+      this.damage += event.amount;
```

Comment:
Remember that absorbed damage is effective damage, this should be `this.damage += event.amount + (event.absorbed || 0);`